android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt

```diff
@@ -0,0 +1,134 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import com.cuhacking.atlas.MainActivity
+import com.cuhacking.atlas.common.SearchResult
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+
+val viewModel = MainActivity().searchViewModel
+val searchResultsVisible = mutableStateOf(false)
+val searchResultsFlow = viewModel.searchResults.flow
+
+@Suppress("MagicNumber")
+@Composable
+fun SearchBar() {
+    var query by remember { mutableStateOf("") }
+    val coroutineScope = rememberCoroutineScope()
+    val focusManager = LocalFocusManager.current
+
+    Surface(
+        elevation = (10.dp),
+        modifier = Modifier
+            .height(75.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 20.dp)
+            .padding(top = 20.dp),
+        shape = RoundedCornerShape(8.dp)
+    ) {
+        TextField(
+            value = query,
+            onValueChange = { textFieldValue: String ->
+                query = textFieldValue
+                coroutineScope.launch {
+                    viewModel.getSearchResults(query)
+                    searchResultsVisible.value = true
+                }
+            },
+            singleLine = true,
+            keyboardActions = KeyboardActions(
+                onSearch = {
+                    coroutineScope.launch {
+                        viewModel.getSearchResults(query)
+                    }
+                    focusManager.clearFocus(true)
+                }
+            ),
+            keyboardOptions = KeyboardOptions(imeAction = ImeAction.Search),
+            leadingIcon = {
+                Icon(Icons.Filled.Search, "Search")
+            },
+            trailingIcon = {
+                if (query.isNotEmpty()) {
+                    Icon(Icons.Filled.Clear, "Clear", modifier = Modifier.clickable(true, onClick = {
+                        query = ""
+                        searchResultsVisible.value = false
+                    }))
+                }
+            },
+            placeholder = { Text("Search...") },
+        )
+    }
+}
+
+@Suppress("MagicNumber")
+@Composable
+fun SearchResultView(searchResult: SearchResult, onClick: () -> Unit) {
+    Card(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 20.dp)
+            .clickable(onClick = onClick)
+            .border(width = 0.5.dp, color = Color.LightGray),
```

Comment:
```suggestion
            .border(width = 0.5.dp, color = MaterialTheme.color.onSurface.copy(alpha = ContentAlpha.disabled)),
```

Reply:
ContentAlpha.disabled makes the results non-clickable, is that what we want?

Reply:
🤔
That's not supposed to happen.

Reply:
actually nvm, might be from another change I made 😆

Reply:
the update to beta 8 changed something, I'll look into it